lib/SIL/SILPrinter.cpp

```diff
@@ -922,6 +926,11 @@ class SILPrinter : public SILInstructionVisitor<SILPrinter> {
     case SILNodeKind::SILUndef:
       printSILUndef(cast<SILUndef>(node));
       return;
+
+    case SILNodeKind::MultipleValueInstructionResult:
+      // We do not print anything for multiple value instruction result since we
+      // print out the result when we print out various instructions.
+      return;
```

Comment:
No.  This may only happen in debugging paths, but just like the SILArgument case, you do need to handle that — this is what will be called if someone calls dump() on a SILValue that happens to be a MultipleValueInstructionResult.

I think the appropriate debug-printing is something like printing the instruction with a special emphasis on the appropriate result, like

```
  %1 = destructure ...
  (%2, **%3**, %4) = destructure ...
```